airflow/executors/kubernetes_executor.py

```diff
@@ -353,12 +353,12 @@ def run_next(self, next_job: KubernetesJobType) -> None:
         self.run_pod_async(pod, **self.kube_config.kube_client_request_args)
         self.log.debug("Kubernetes Job created!")
 
-    def delete_pod(self, pod_id: str, namespace: str) -> None:
-        """Deletes POD."""
+    def delete_pod(self, pod_name: str, namespace: str) -> None:
```

Comment:
I guess arguably this is part of the public api?

Reply:
This is an interesting grey area. We say in the new public interface docs that executors are public and we even encourage extending them:
> Airflow has a set of Executors that are considered public. You are free to extend their functionality

But this is not the `KubernetesExecutor` class itself that you're modifying it's the job watcher singleton (which is indirectly used by the executor class via the scheduler class, which is itself passed in as a parameter to the executor).

So I could see this being a private class who's interface we don't guarantee to be consistent release to release.

I'm interested to hear what others think though.

Reply:
> I'm interested to hear what others think though.

me too. i commented on the discussion thread.  ideally i think only base executor is public -- i.e. the executor interface --  and instances are not.

Reply:
I commented in the thread too :) https://lists.apache.org/thread/xv5p733j2romrhsl174fpnvfb8fw30co

Reply:
Outcome: it's not part of the public api 👍